scheduler/server/load_based_sched_alg.go

```diff
@@ -111,10 +112,10 @@ func NewLoadBasedAlg(config *LoadBasedAlgConfig, tasksByJobClassAndStartTimeSec
 	return lbs
 }
 
-// jobWaitingTaskIds map waiting task ids to the job state objects
-type jobWaitingTaskIds struct {
-	jobState       *jobState
-	waitingTaskIDs []string
+// jobWaitingTasks waiting task ids (in the order they should be started) for a job
+type jobWaitingTasks struct {
+	jobState     *jobState
+	waitingTasks []*taskState
```

Comment:
we used to have taskID in waitingTaskIDs, and use that to get the *taskState from the jobState's notRunning[] map.  Instead we now just store the *taskState's for tasks not started in waitingTasks (in the order they should be started) (see changes below)